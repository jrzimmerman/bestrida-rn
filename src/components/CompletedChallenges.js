import React from 'react';
import {
  Image,
  ListView,
  StatusBar,
  Text,
  TouchableOpacity,
  View
} from 'react-native';
import styles from './styles';
import CompletedChallengeDetail from './CompletedChallengeDetail';

const stravaProfilePic = require('../images/strava_profile_pic.png');

class CompletedChallenges extends React.Component {
  constructor(props) {
    super(props);
    const ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 });
    this.state = {
      dataSource: ds.cloneWithRows([
        'John', 'Joel', 'James', 'Jimmy', 'Jackson', 'Jillian', 'Julie', 'Devin'
      ])
    };
    this.handlePress = this.handlePress.bind(this);
  }

  handlePress() {
    this.props.navigator.push({ component: CompletedChallengeDetail });
  }

  render() {
    return (
      <View style={styles.container}>
        <StatusBar barStyle="light-content" />
        <ListView
          style={styles.list}
          dataSource={this.state.dataSource}
          renderRow={(rowData) => (
            <TouchableOpacity onPress={this.handlePress} style={styles.row}>
              <View style={styles.challengeImageView}>
                <Image
                  style={styles.challengeImage}
                  source={stravaProfilePic}
                />
              </View>
              <View style={styles.challengeDetail}>
                <Text style={styles.challengeText}>Opponent: OPPONENT</Text>
                <Text style={styles.challengeText}>Segment: SEGMENT NAME</Text>
                <Text style={styles.challengeText}>You won this challenge!</Text>
              </View>
            </TouchableOpacity>
          )}
        />
      </View>
    );
  }
}

export default CompletedChallenges;
