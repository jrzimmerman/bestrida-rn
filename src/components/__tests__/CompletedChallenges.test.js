import 'react-native';
import React from 'react';
import { CompletedChallenges } from '../CompletedChallenges';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders CompletedChallenges component', () => {
  const completed = {
    loading: false,
    challenges: [],
    error: null
  };
  const tree = renderer.create(
    <CompletedChallenges
      completed={completed}
      dispatch={fn => fn}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
