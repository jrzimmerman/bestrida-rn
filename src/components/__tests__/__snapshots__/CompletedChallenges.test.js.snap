// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CompletedChallenges component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2B2B2B",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    dataSource={
      ListViewDataSource {
        "items": 0,
      }
    }
    enableEmptySections={true}
    initialRows={10}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    removeClippedSubviews={false}
    renderRow={[Function]}
    renderScrollComponent={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "marginTop": 5,
      }
    }
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</View>
`;
